core/application/bloodDonationWorkflow/DonorSearchService.ts

```diff
@@ -0,0 +1,181 @@
+      }
+      await donorSearchRepository.update({
+        ...donorRoutingAttributes,
+        id: requestPostId,
+        status: DonationStatus.PENDING,
+        retryCount: 0
+      })
+    }
+
+    const retryCount = donorSearchRecord?.retryCount ?? 0
+    const updatedRecord: Partial<DonorSearchDTO> = {
+      ...donorRoutingAttributes,
+      id: requestPostId,
+      retryCount: retryCount + 1
+    }
+    const hasRetryCountExceeded = retryCount >= Number(process.env.MAX_RETRY_COUNT)
+    if (hasRetryCountExceeded) {
+      updatedRecord.status = DonationStatus.COMPLETED
+      await donorSearchRepository.update(updatedRecord)
+      return
+    }
+
+    await donorSearchRepository.update(updatedRecord)
+
+    const stepFunctionPayload: StepFunctionInput = {
+      seekerId,
+      requestPostId,
+      createdAt,
+      donationDateTime: donorRoutingAttributes.donationDateTime,
+      requestedBloodGroup: donorRoutingAttributes.requestedBloodGroup,
+      bloodQuantity: Number(donorRoutingAttributes.bloodQuantity),
+      urgencyLevel: donorRoutingAttributes.urgencyLevel,
+      geohash: donorRoutingAttributes.geohash,
+      seekerName: userProfile.name,
+      patientName: donorRoutingAttributes.patientName ?? '',
+      location: donorRoutingAttributes.location,
+      contactNumber: donorRoutingAttributes.contactNumber,
+      transportationInfo: donorRoutingAttributes.transportationInfo ?? '',
+      shortDescription: donorRoutingAttributes.shortDescription ?? '',
+      city: donorRoutingAttributes.city,
+      retryCount: retryCount + 1,
+      message: getBloodRequestMessage(
+        donorRoutingAttributes.urgencyLevel,
+        donorRoutingAttributes.requestedBloodGroup,
+        donorRoutingAttributes.shortDescription ?? ''
+      )
+    }
+
+    await stepFunctionModel.startExecution(
+      stepFunctionPayload,
+      `${requestPostId}-${donorRoutingAttributes.city}-(${donorRoutingAttributes.requestedBloodGroup})-${Math.floor(Date.now() / 1000)}`
+    )
+  }
+
+  async getDonorSearch(
+    seekerId: string,
+    createdAt: string,
+    requestPostId: string,
+    donorSearchRepository: Repository<DonorSearchDTO>
+  ): Promise<DonorSearchDTO> {
+    const donorSearchRecord = await donorSearchRepository.getItem(
+      `${DONOR_SEARCH_PK_PREFIX}#${seekerId}`,
+      `${DONOR_SEARCH_PK_PREFIX}#${createdAt}#${requestPostId}`
+    )
+    if (donorSearchRecord === null) {
+      throw new Error('Donor search record not found.')
+    }
+    return donorSearchRecord
+  }
+
+  async updateDonorSearch(
+    seekerId: string,
+    createdAt: string,
+    requestPostId: string,
+    geohash: string,
+    remainingGeohashesToProcess: string[],
+    currentNeighborSearchLevel: number,
+    donorSearchRepository: Repository<DonorSearchDTO>
+  ): Promise<void> {
+    const updatedRecord: Partial<DonorSearchDTO> = {
+      id: requestPostId,
+      seekerId,
+      createdAt
+    }
+    if (remainingGeohashesToProcess.length === 0) {
```

Comment:
Bhai if we use the ternary operator like below code may be look more readable and clean.

```
  const updatedRecord: Partial<DonorSearchDTO> = {
    id: requestPostId,
    seekerId,
    createdAt,
    currentNeighborSearchLevel: 
      remainingGeohashesToProcess.length === 0 
        ? currentNeighborSearchLevel + 1 
        : currentNeighborSearchLevel,
    remainingGeohashesToProcess: 
      remainingGeohashesToProcess.length === 0 
        ? getGeohashNthNeighbors(geohash, currentNeighborSearchLevel + 1) 
        : remainingGeohashesToProcess,
  };
```

Reply:
needed to update the existing code. not sure if its possible now. please check again

Reply:
Updated